packages/spidermonkey.rb

```diff
@@ -0,0 +1,62 @@
+require 'package'
+
+class Spidermonkey < Package
+  description 'SpiderMonkey is Mozilla\'s JavaScript engine written in C and C++.'
+  homepage 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey'
+  version '84.0'
+  compatibility 'all'
+  source_url 'https://archive.mozilla.org/pub/firefox/releases/84.0/source/firefox-84.0.source.tar.xz'
+  source_sha256 '23273ef0165b243f5d0908c38e7854d38070282c9b526e8d93b7503cd5f69138'
+
+  depends_on 'yasm'
+  depends_on 'rust' => :build
+  
+  def self.prebuild
+    puts
+    puts "Building spidermonkey requires at least 6.4 GB of space available on /mnt/stateful_partition.".lightblue
+    puts "It WILL fail without this space. Press Ctrl + C to cancel.".lightblue
+    puts
+    sleep(5)
+  end
+
+  case ARCH
+    when 'aarch64', 'armv7l'
+     @gnuabi = 'gnueabihf'
+    when 'x86_64', 'i686'
+      @gnuabi = 'gnu'
+  end
+
+  def self.build
+    ENV['LC_ALL'] = 'C' # https://bugs.gentoo.org/746215
+    ENV['MACH_USE_SYSTEM_PYTHON'] = "#{CREW_PREFIX}/bin/python3"
+    ENV['CC'] = 'gcc' # Defaults to clang, but configure.py complains about oudated libstdc++ when using clang
+    ENV['CXX'] = 'g++'
+    ENV['AR'] = 'ar'
+    ENV['AS'] = 'as'
+    ENV['RANLIB'] = 'ranlib'
+    ENV['NM'] = 'nm'
+    ENV['LD'] = 'ld.gold'
```

Comment:
We shouldn't need all these environment variables I'm thinking.

Reply:
Also if gold linker is needed, it can be invoked with ```env CFLAGS='-fuse-ld=gold' CXXFLAGS='-fuse-ld=gold' ./configure...```